phys2bids/physio_obj.py

```diff
@@ -0,0 +1,321 @@
+                print(f'Found {tps_missing} tps less than expected!')
+                if tr:
+                    print('Correcting time offset, assuming missing tps '
+                          'are at the beginning (try again with '
+                          'a more liberal thr')
+                    time_offset -= (tps_missing * tr)
+                else:
+                    print('Can\'t correct time offset, (try again specifying '
+                          'tr or with a more liberal thr')
+
+            else:
+                print('Found just the right amount of tps!')
+
+        else:
+            print('Cannot check the number of tps')
+
+        cls.timeseries[0] -= time_offset
+        cls.num_tps_found = num_tps_found
+
+
+class blueprint_output():
+    """
+    Main output object for phys2bids.
+    Contains the blueprint to be exported.
+
+    Properties - Input
+    ------------------
+    timeseries : (ch x tps) :obj:`numpy.ndarray`
+        Numpy 2d array of timeseries
+        Contains all the timeseries recorded.
+        Impose same frequency!
+    freq : float
+        Shared frequency of the object.
+            Properties
+    ch_name : (ch) list of strings
+        List of names of the channels - can be the header of the columns
+        in the output files.
+    units : (ch) list of strings
+        List of the units of the channels.
+    start_time : float
+        Starting time of acquisition (equivalent to first TR,
+        or to the opposite sign of the time offset).
+
+    Methods
+    -------
+    return_index:
+        Returns the proper list entry of all the
+        properties of the object, given an index.
+    delete_at_index:
+        Returns all the proper list entry of the
+        properties of the object, given an index.
+    init_from_blueprint:
+        method to populate from input blueprint instead of init
+    """
+    def __init__(self, timeseries, freq, ch_name, units, start_time):
+        self.timeseries = is_valid(timeseries, np.ndarray)
+        self.ch_amount = self.timeseries.shape[0]
+        self.freq = is_valid(freq, (int, float))
+        self.ch_name = has_size(ch_name, self.ch_amount, 'unkown')
+        self.units = has_size(units, self.ch_amount, '[]')
+        self.start_time = start_time
+
+    def return_index(cls, idx):
+        """
+        Returns all the proper list entry of the
+        properties of the object, given an index.
+
+        Input
+        -----
+        idx: int
+            Index of elements to return
+        """
+        return (cls.timeseries[idx], cls.ch_amount, cls.freq,
+                cls.ch_name[idx], cls.units[idx], cls.start_time)
+
+    def delete_at_index(cls, idx):
+        """
+        Returns all the proper list entry of the
+        properties of the object, given an index.
+
+        Input
+        -----
+        idx: int or range
+            Index of elements to delete from all lists
```

Comment:
You should comment on the output.

Reply:
It doesn't have an output per se, but an outcome.
Should we write outcomes?

Reply:
I'd say we should always comment on whatever we return.

Reply:
We don't return anything!

Reply:
I'm sorry, this was meant for line 289.

Reply:
In any case, it's updated with "Outcomes"